projects/kit/components/combo-box/combo-box.component.ts

```diff
@@ -167,10 +167,11 @@ export class TuiComboBoxComponent<T>
     }
 
     handleOption(item: T) {
-        this.focusInput();
+        this.setNativeValue(this.stringify(item));
         this.close();
         this.updateSearch(null);
         this.updateValue(item);
+        this.focusInput();
```

Comment:
Let's move it right before `this.close()`. Let's not change order of stuff, it might be relevant :)